connectors/kafka/src/main/java/forklift/connectors/KafkaConnector.java

```diff
@@ -0,0 +1,133 @@
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Manages both consuming and producing events on the kafka message broker.
+ */
+public class KafkaConnector implements ForkliftConnectorI {
+    private static final Logger log = LoggerFactory.getLogger(KafkaConnector.class);
+    private final String kafkaHosts;
+    private final String schemaRegistries;
+    private final String groupId;
+    private KafkaProducer<?, ?> kafkaProducer;
+    private MessageStream messageStream = new MessageStream();
+    private KafkaController controller;
+
+    /**
+     * Constructs a nw instance of the KafkaConnector
+     *
+     * @param kafkaHosts       list of kafka servers in host:port,... format
+     * @param schemaRegistries list of schema registry servers in http://host:port,... format
+     * @param groupId          the groupId to use when subscribing to topics
+     */
+    public KafkaConnector(String kafkaHosts, String schemaRegistries, String groupId) {
+        this.kafkaHosts = kafkaHosts;
+        this.schemaRegistries = schemaRegistries;
+        this.groupId = groupId;
+    }
+
+    @Override
+    public void start() throws ConnectorException {
+        //We do nothing here.  Consumer and producer are created when needed
+    }
+
+    private KafkaProducer createKafkaProducer() {
+
+        Properties producerProperties = new Properties();
+        producerProperties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaHosts);
+        producerProperties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+                               io.confluent.kafka.serializers.KafkaAvroSerializer.class);
+        producerProperties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+                               io.confluent.kafka.serializers.KafkaAvroSerializer.class);
+        //schema.registry.url is a comma separated list of urls
+        producerProperties.put("schema.registry.url", schemaRegistries);
+        return new KafkaProducer(producerProperties);
+    }
+
+    private KafkaController createController() {
+        Properties props = new Properties();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaHosts);
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+        props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, io.confluent.kafka.serializers.KafkaAvroDeserializer.class);
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, io.confluent.kafka.serializers.KafkaAvroDeserializer.class);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 200);
+        props.put("schema.registry.url", schemaRegistries);
+        props.put("specific.avro.reader", false);
```

Comment:
The property name for this is `KafkaAvroDeserializerConfig.SPECIFIC_AVRO_READER_CONFIG`.


Default value is `false`, but either way is okay. :man_shrugging: